src/main/scala/utils/OsmoGridUtils.scala

```diff
@@ -139,36 +139,14 @@ object OsmoGridUtils {
     * @return
     *   a list of all unique connections
     */
-  def getAllUniqueCombinations[T](
-      nodes: List[T]
-  ): List[(T, T)] = {
+  def getAllUniqueCombinations[T](nodes: List[T]): List[(T, T)] = {
     if (nodes.size < 2) {
       List.empty
-    } else if (nodes.size == 2) {
-      List((nodes(0), nodes(1)))
     } else {
-      val connections: util.List[(T, T)] =
-        new util.ArrayList[(T, T)]
-
-      // algorithm to find all unique combinations
-      nodes.foreach(nodeA => {
-        nodes.foreach(nodeB => {
-          // it makes no sense to connect a node to itself => nodeA and nodeB cannot be the same
-          if (nodeA != nodeB) {
-            // two combinations possible
-            val t1 = (nodeA, nodeB)
-            val t2 = (nodeB, nodeA)
-
-            // if none of the combinations is already added, the first combination is added
-            if (!connections.contains(t1) && !connections.contains(t2)) {
-              connections.add(t1)
-            }
-          }
-        })
-      })
-
-      // returns all unique connections
-      connections.asScala.toList
+      for {
+        (a, i) <- nodes.zipWithIndex
+        b <- nodes.drop(i + 1)
```

Comment:
This look fine to me, would be nice to have some clarifying commentary here because it does not seem to be self-explanatory to many. 
Maybe something like:
// Loops through all nodes as a and combines each node a with each b of the same nodes list.
// Nodes as b are restricted to those that occurr later in the list than each node a respectively.

(the commentary might not help much, but might at least give hints as to what is going on here)